README.tmpl.md

```diff
@@ -0,0 +1,206 @@
+them to `.words`. If `gospel` is run again, it will accept the all words in
+the source tree as correctly spelled as they now appear in the projects
+dictionary. Since hunspell dictionaries are just text files, you can go
+through the list of words to remove properly identified misspellings and
+leave in words that are domain-specific and incorrectly marked.
+
+For example the `.words` file at the root of the `gospel` repo includes
+words that would otherwise be flagged with the default "en_US" spelling
+dictionary and was autogenerated using the command above.
+
+```
+{{.words -}}
+```
+
+This file can also be edited to make use of more advanced hunspell
+features. See [Hunspell Dictionaries](#hunspell-dictionaries) below.
+
+
+## Command Line Options
+
+`gospel` provides a number of command line options. The full list can
+be obtained by running `gospel -h`, but the majority correspond directly
+with `.gospel.conf` options and are described [there](.gospel.conf).
+
+The remaining options are not intended to be persistently stored:
+
+- `-config` — whether to use config file (default true, intended for debugging use).
+- `-dict-paths` — a colon-separated directory list containing hunspell dictionaries (defaults to a system-specific value).
+- `-misspellings` — a file path to write a dictionary of misspellings to (see [Work Flow](#work-flow) above).
+- `-since` — a git ref specifying that only changes since then should be considered for misspelling (requires git).
+- `-update-dict` — whether the `-misspellings` flag is being used to update a dictionary that already exists.
+- `-write-config` — emit a config file based on flags and existing config to stdout and exit.
+
+
+## Configuration Files
+
+`gospel` uses two configuration file types, `.words` files at the module
+roots of packages that are being checked, and a `.gospel.conf` file at the
+module root of the module in which `gospel` was invoked.
+
+If `gospel` is being used in CI, these files should be committed to the
+repository.
+
+
+### `.words`
+
+The `.words` file is a hunspell `.dic` formatted file. At its simplest, this
+is a plain text file with a list of words, one word per line and an initial
+word count on the first line. The word count is a hint only and does not need
+to match the number of words in the dictionary (a value of `1` will work fine
+if you are constructing the file by hand).
+
+`gospel` will read _all_ `.words` files found at module roots in for the
+packages that are being checked and build a dictionary from them to give to
+hunspell for the spelling checks.
+
+Hunspell dictionaries are able to express more than just word matches though,
+and are able to indicate some grammatically related sets of words based on
+rules. This is covered lightly [below](#hunspell-dictionaries).
+
+
+### `.gospel.conf`
+
+Runtime behaviour of `gospel` can be modified in a persistent way through the
+TOML format `.gospel.conf` file. A number of options are provide:
```

Comment:
```suggestion
TOML format `.gospel.conf` file. A number of options are provided:
```

Reply:
I guess gospel isn't a full-on English language checker, so it doesn't realise that `is <verb>` should likely be `is <adjective>`.

Reply:
Thanks. No grammar parsing is hard. Spelling checkers are essentially just scanners without any grammar awareness.